src/gridcoin/scraper/scraper_net.h

```diff
@@ -127,41 +142,40 @@ class CScraperManifest
 
 
 public: /*==== fields ====*/
-
-    const uint256* phash;
-    std::string sCManifestName;
-    CPubKey pubkey;
-    std::vector<unsigned char> signature;
+    const uint256* phash GUARDED_BY(cs_manifest);
+    std::string sCManifestName GUARDED_BY(cs_manifest);
+    CPubKey pubkey GUARDED_BY(cs_manifest);
+    std::vector<unsigned char> signature GUARDED_BY(cs_manifest);
 
     struct dentry {
-        std::string project;
-        std::string ETag;
-        unsigned int LastModified =0;
-        int part1 =-1;
-        unsigned partc =0;
-        int GridcoinTeamID =-1;
-        bool current =0;
-        bool last =0;
+        std::string project GUARDED_BY(cs_manifest);
+        std::string ETag GUARDED_BY(cs_manifest);
+        unsigned int LastModified GUARDED_BY(cs_manifest) = 0;
+        int part1 GUARDED_BY(cs_manifest) = -1;
+        unsigned partc GUARDED_BY(cs_manifest) = 0;
+        int GridcoinTeamID GUARDED_BY(cs_manifest) = -1;
+        bool current GUARDED_BY(cs_manifest) = 0;
+        bool last GUARDED_BY(cs_manifest) = 0;
```

Comment:
The `GUARDED_BY` attribute won't work here. The `dentry` class is related to `CScraperManifest` by name only&mdash;it is legal to instantiate these alone, so static analysis cannot assume that a `CScraperManifest` object will exist in scope to validate locks on `cs_manifest`. `CScraperManifest::projects` is the field we care about, and that one has the attribute already.

Reply:
Aha! Thanks :)

Reply:
Moving the scraper work to another PR